### PR TITLE
Initial GHA implementation

### PR DESCRIPTION
**Pull request by leezen**
Related to #98

### CODE REVIEW

**Review by jaxxstorm — APPROVED** [review]
Amazing! 🚢 it!

File: .github/workflows/master.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+name: Staging
```
Comment (stack72) [inline .github/workflows/master.yml:1]: Nit on this title :D

Comment (stack72) [reply .github/workflows/master.yml:1]: How does staging relate to "master"

File: .github/workflows/pull-request.yml [hunk 2]
```diff
@@ -35,3 +32,31 @@ jobs:
...
+      uses: aws-actions/configure-aws-credentials@v1
+      with:
+        aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        aws-region: us-west-2
+        role-to-assume: ${{ env.AWS_ROLE_TO_ASSUME }}
+        role-external-id: ${{ env.PULUMI_STACK_NAME }}
+        role-duration-seconds: 3600
+        role-session-name: get.pulumi.com-${{ github.run_id }}
+    - uses: pulumi/actions@v1
```
Comment (stack72) [inline .github/workflows/pull-request.yml:57]: total candidate to use the new GHA soon :)

**Review by stack72 — APPROVED** [review]

**Comment by stack72** [comment]
This is a great addition to move us to GHA BUT this will fail because of credentials when a PR is sent so we are going to need to have a second set of work to adapt to the command dispatch style approach we have in templates and examples

I am happy to take this initial work and extend it

File: .github/workflows/pull-request.yml (hunk 2 above)
Comment (leezen) [reply .github/workflows/pull-request.yml:57]: Yup -- agreed 😄

File: .github/workflows/master.yml (hunk 1 above)
Comment (leezen) [reply .github/workflows/master.yml:1]: The `staging` stack (which the `master` branch deploys to) is hosted at https://get.pulumi-staging.io/ -- which is why I called the workflow `Staging`. Do you feel like this is the wrong name and we should stick w/ the branch name?

**Comment by leezen** [comment]
I made a few changes here and there to get this actually working in GHA after having some issues w/ the tokens and whatnot. Would appreciate another look if you don't mind.

**Review by stack72 — APPROVED** [review]
This is awesome! New GHA champion :)

**Comment by stack72** [comment]
:shipit:

**Pull request merged**